lib/discourse_h_captcha/create_users_controller_patch.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module DiscourseHCaptcha
+  module CreateUsersControllerPatch
+    H_CAPTCHA_VERIFICATION_URL = "https://hcaptcha.com/siteverify".freeze
+
+    extend ActiveSupport::Concern
+    included { before_action :check_h_captcha, only: [:create] }
+
+    def check_h_captcha
+      h_captcha_token = fetch_h_captcha_token
+      raise Discourse::InvalidAccess.new if h_captcha_token.blank?
+
+      response = send_h_captcha_verification(h_captcha_token)
+
+      validate_h_captcha_response(response)
+    rescue => e
+      Rails.logger.warn("Error parsing hCaptcha response: #{e}")
+      fail_with("h_captcha_verification_failed")
+    end
+
+    private
+
+    def send_h_captcha_verification(h_captcha_token)
+      uri = URI.parse(H_CAPTCHA_VERIFICATION_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
```

Comment:
We should use `FinalDestination::HTTP` for SSRF protection. (not a massive concern here since the domain is hard-coded, but still good practice)